conjureplugin/conjureplugin.go

```diff
@@ -51,23 +51,57 @@ func Run(params ConjureProjectParams, verify bool, projectDir string, stdout io.
 			GenerateCLI:          currParam.CLI,
 			GenerateFuncsVisitor: currParam.AcceptFuncs,
 		}
+
+		files, err := conjure.GenerateOutputFiles(conjureDef, outputConf)
+		if err != nil {
+			return errors.Wrap(err, "failed to generate conjure output files")
+		}
+
 		if verify {
-			diff, err := diffOnDisk(conjureDef, projectDir, outputConf)
+			diff, err := diffOnDisk(projectDir, files)
 			if err != nil {
 				return err
 			}
+
+			var msg string
 			if len(diff.Diffs) > 0 {
-				verifyFailedFn(k, diff.String())
+				msg = diff.String()
+			}
+
+			if !currParam.SkipDeleteGeneratedFiles {
+				filesToDelete, err := computeObsoleteFiles(outputConf.OutputDir, files)
+				if err != nil {
+					return err
+				}
```

Comment:
Lines 71-75 and 90-93 are basically duplicated -- seems like it would make more sense to have:

```
var filesToDelete []string
if !currParam.SkipDeleteGeneratedFiles {
				filesToDelete, err = computeObsoleteFiles(outputConf.OutputDir, files)
				if err != nil {
					return err
				}
}
```

As shared logic before the `if/else`, and then the `len(filesToDelete) > 0` and `for ... := range filesToDelete` would both naturally be noops if the slice is empty.